x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/lookup/ExpressionQueryList.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.operator.lookup;
+
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+
+import java.util.List;
+
+/**
+ * A {@link LookupEnrichQueryGenerator} that combines multiple {@link QueryList}s into a single query.
+ * Each query in the resulting query will be a conjunction of all queries from the input lists at the same position.
+ * In the future we can extend this to support more complex expressions, such as disjunctions or negations.
+ */
+public class ExpressionQueryList implements LookupEnrichQueryGenerator {
```

Comment:
I wonder if we should flip these into methods on the `Expression`. Or maybe some `interface` that we implement on types that can make the queries.

Reply:
It's `TranslationAware` - or, well, almost the same.

Reply:
Anyway, I think that's a potential follow-up to this. This code does fine for now and we can flip later.

Reply:
Er, well, it's not the same as `TranslationAware`. But it's similar!

Reply:
:+1:

It's a for later thing anyway.